dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -116,6 +116,32 @@ def manual_pass
     redirect_to :manual_pass_form
   end
 
+  # get /admin/user_progress
+  def user_progress_form
+    user_identifier = params[:user_identifier]
+    offset = params[:offset] || 0 # Not currently exposed in admin UI but can be manually added to URL
```

Comment:
Nit: since there are two groups of data being returned, user_levels and user_scripts it might be worth naming offset to `script_offset` for clarity on which data is being paged